site/source/docs/debugging/Sanitizers.rst

```diff
@@ -378,3 +372,56 @@ option ``malloc_context_size=0``, like this:
 This prevents ASan from reporting the location of memory leaks, as well as
 offer insight into where the memory for a heap-based memory error originated,
 but may provide tremendous speed ups.
+
+Comparison to ``SAFE_HEAP``
```

Comment:
Thanks, I like these explanations a lot!

One thing I feel is missing is that you mention JavScript-specific issues below (e.g., in particular, getting `undefined` out of a Typed Array) but it's not clear what kind of JavaScript this is in relation to / why is it relevant.

Is it about asm.js (and so only relevant for those who target it)? Or can it catch issues in user snippets?

Some additional note and/or examples might be helpful for the context.

Reply:
Very good point, thanks! Added notes now. This is relevant for all the JS, which means all the support code, and also the compiled code when using wasm2js.

Reply:
> This is relevant for all the JS

Oh interesting, even things like EM_ASM / EM_JS snippets? Is this done by post-processing those with Acorn plugins or something like that?

Reply:
Yes, EM_ASM/EM_JS ends up with all the rest of the JS that the optimizer sees. The acorn-based optimizer then post-processes them.

Reply:
That's pretty cool, I didn't know that actually :)